src/components/articlePage/articleHeader.tsx

```diff
@@ -6,35 +6,47 @@ import profileIcon from '@/public/icons/profile_icon.png';
 import LikeButton from '@/components/common/likeButton/likeButton';
 import ActionMenu from '../community/actionMenu/actionMenu';
 import { useRouter } from 'next/navigation';
-import { useMutation, useQueryClient } from '@tanstack/react-query';
-import { deleteArticle } from '@/services/api/article';
+import { useDeleteArticleMutation } from '@/hooks/articles/useDeleteArticleMutation';
+import { useSetAtom } from 'jotai';
+import { confirmModalAtom } from '@/lib/store/modalAtoms';
+import { useMe } from '@/hooks/useMe';
+import { useMessageModal } from '@/hooks/modals/useMessageModal';
 
 export default function ArticleHeader({
   id,
   nickname,
   title,
   createdAt,
+  likeCount,
+  isLiked,
+  ownerId,
 }: ArticleHeaderProps) {
-  const queryClient = useQueryClient();
+  const deleteArticleMutation = useDeleteArticleMutation(id);
+  const setConfirmModalState = useSetAtom(confirmModalAtom);
+  const { setMessage } = useMessageModal();
   const router = useRouter();
 
-  const deleteArticleMutation = useMutation({
-    mutationFn: () => deleteArticle(id),
-    onSuccess: () => {
-      queryClient.invalidateQueries({ queryKey: ['bestArticles'] });
-      queryClient.invalidateQueries({ queryKey: ['normalArticles'] });
-      router.push('/community');
-    },
-  });
+  const { data: me } = useMe();
+  const onEditButtonClick = () => {
+    if (me?.id !== ownerId)
+      return setMessage('본인의 게시물만 수정할 수 있습니다.');
+    router.push(`/community/${id}/edit`);
+  };
 
   return (
     <div className='flex flex-col mb-4 md:mb-4 xl:mb-6 w-full'>
       <div className='flex w-full mb-4 items-center justify-between'>
         <h2 className='font-bold text-xl'>{title}</h2>
         <ActionMenu
           id={id}
-          onEditButtonClick={() => router.push(`/community/${id}/edit`)}
-          onDeleteButtonClick={() => deleteArticleMutation.mutate()}
+          onEditButtonClick={() => onEditButtonClick()}
```

Comment:
```suggestion
          onEditButtonClick={onEditButtonClick}
```

아마 이런 케이스는 자동으로 이렇게 바꿔주는 eslint 룰이 있었던 걸로 기억하는데, 한번 찾아보셔도 좋을 것 같네요.